--- a/plt-5.0/src/racket/gc2/sighand.c	2010-07-04 11:47:22.000000000 +0400
+++ b/plt-5.0/src/racket/gc2/sighand.c	2010-07-04 11:48:09.000000000 +0400
@@ -55,7 +55,11 @@
       /* supposedly its coming from the user via kill */
       /* so just ignore it. */
       printf("SIGSEGV SI_USER SI_CODE %i fault on addr %p\n", c, p);
+      #ifdef MZ_USE_PLACES
       printf("pid %i uid %i thread %lx\n", si->si_pid, si->si_uid, mz_proc_thread_self());
+      #else
+      printf("pid %i uid %i\n", si->si_pid, si->si_uid);
+      #endif      
       return;
     }
     if (c == 128 ) {
